standardization/src/test/scala/za/co/absa/enceladus/standardization/interpreter/StandardizationInterpreter_DecimalSuite.scala

```diff
@@ -0,0 +1,144 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.standardization.interpreter
+
+import java.text.{DecimalFormat, NumberFormat}
+import java.util.Locale
+import org.apache.spark.sql.types.{DecimalType, StringType, StructField, StructType}
+import org.scalatest.FunSuite
+import za.co.absa.enceladus.utils.error.{ErrorMessage, UDFLibrary}
+import za.co.absa.enceladus.utils.testUtils.SparkTestBase
+
+class StandardizationInterpreter_DecimalSuite extends FunSuite with SparkTestBase {
+  import spark.implicits._
+
+  private implicit val udfLib: UDFLibrary = new UDFLibrary
+
+  private val desiredSchema = StructType(Seq(
+    StructField("description", StringType, nullable = false),
+    StructField("small", DecimalType(5,2), nullable = false),
+    StructField("big", DecimalType(38,18), nullable = true)
+  ))
+
+  private val zero = BigDecimal("0E-18")
+  private val bigDecimalFormat = {
+    val pattern = "0.000000000000000000" //18 decimal places
+    val nf = NumberFormat.getNumberInstance(Locale.US)
+    val df = nf.asInstanceOf[DecimalFormat]
+    df.applyPattern(pattern)
+    df
+  }
+
+  private def bd(number: Double): BigDecimal = {
+    val s: String =  bigDecimalFormat.format(number)
+    BigDecimal(s)
+  }
+
+  test("From String") {
+    val seq = Seq(
+      ("01-Pi", "3.14", "3.14"),
+      ("02-Null", null, null),
+      ("03-Long", Long.MaxValue.toString, Long.MinValue.toString),
+      ("04-infinity", "-Infinity", "Infinity"),
+      ("05-Really big", "123456789123456791245678912324789123456789123456789.12",
+        "12345678912345679124567891232478912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912346789123456789123456789123456789123456791245678912324789123456789123456789123456789123456789123456791245678912324789123456789123456789123456789123456789123456789123456789123456789.1"),
+      ("06-Text", "foo", "bar"),
+      ("07-Exponential notation", "-1.23E2", "+9.8765E-4"),
+      ("08-Small overflow", "1000", "1000"),
+      ("09-Loss of precision", "123.456", "123.456")
+    )
+    val src = seq.toDF("description","small", "big")
+    showDateFrame(src)
+
+    val std = StandardizationInterpreter.standardize(src, desiredSchema, "").cache()
+    showDateFrame(std)
+
+    val exp = Seq(
+      DecimalRow("01-Pi", Option(bd(3.14)), Option(bd(3.14))),
+      DecimalRow("02-Null", Option(zero), None, Seq(
+        ErrorMessage("stdNullError", "E00002", "Standardization Error - Null detected in non-nullable attribute", "small", Seq("null")))),
```

Comment:
Maybe it's time to extract constants for these Error codes. Too much repetition in all strings I'd say.

Reply:
Thanks, actually the `ErrorMessage` companion object has a methods for these standard type ones. I am going to use those.